docs/docs/custom_preconditioners.md

```diff
@@ -0,0 +1,142 @@
+
+# Defining custom gradient preconditioners
+
+NetKet calls _gradient preconditioner_ that class of techniques that transform the
+gradient of the cost function in such a way to improve convergence properties before
+passing it to an optimiser like :ref:`Sgd` or :ref:`Adam`.
+
+Examples of _gradient preconditioners_ are the [Stochastic Reconfiguration](https://www.attaccalite.com/PhDThesis/html/node15.html) method  (also known as [Natural Gradient](https://wiseodd.github.io/techblog/2018/03/14/natural-gradient/), the Linear Method or second order Hessian-based optimisation.
+
+We call those methods _gradient preconditioners_ because they take as input the gradient of the cost function (usually the energy) and some additional information, and output a transformed gradient.
+
+NetKet comes out of the box with Stocastic Reconfiguration (:ref:`netket.optimiser.SR`), but it is possible to define your own method. 
+If you follow the API outlined in this document, you will be able to use your own preconditioner within netket optimisation drivers without issues.
```

Comment:
```suggestion
```suggestion
In the current version, NetKet provides Stocastic Reconfiguration (:ref:`netket.optimiser.SR`) as a built-in method. It is also possible to define your own method. If you implement the API as outlined in this document, you will be able to use your own preconditioner for use in NetKet optimisation driver without issues.
```